lms/djangoapps/oauth_dispatch/views.py

```diff
@@ -0,0 +1,89 @@
+"""
+Views that dispatch processing of OAuth requests to django-oauth2-provider or
+django-oauth-toolkit as appropriate.
+"""
+
+from __future__ import unicode_literals
+
+from django.views.generic import View
+from edx_oauth2_provider import views as dop_views  # django-oauth2-provider views
+from oauth2_provider import models as dot_models, views as dot_views  # django-oauth-toolkit
+
+from auth_exchange import views as auth_exchange_views
+
+from . import adapters
+
+
+class _DispatchingView(View):
+    """
+    Base class that route views to the appropriate provider view.  The default
+    behavior routes based on client_id, but this can be overridden by redefining
+    `select_backend()` if particular views need different behavior.
+    """
+    # pylint: disable=no-member
+
+    dot_adapter = adapters.DOTAdapter()
+    dop_adapter = adapters.DOPAdapter()
+
+    def dispatch(self, request, *args, **kwargs):
+        """
+        Dispatch the request to the selected backend's view.
+        """
+        backend = self.select_backend(request)
+        view = self.get_view_for_backend(backend)
+        return view(request, *args, **kwargs)
+
+    def select_backend(self, request):
+        """
+        Given a request that specifies an oauth `client_id`, return the adapter
+        for the appropriate OAuth handling library.  If the client_id is found
+        in a django-oauth-toolkit (DOT) Application, use the DOT adapter,
+        otherwise use the django-oauth2-provider (DOP) adapter, and allow the
+        calls to fail normally if the client does not exist.
+        """
+
+        if dot_models.Application.objects.filter(client_id=self._get_client_id(request)).exists():
```

Comment:
No.  If no client_id is available, the dop_adapter will be used, and will return the same error it would have if dispatching weren't being used.